### PR TITLE
man containers-registries.conf: clarify mirror association

### PR DESCRIPTION
**Pull request by vrothberg**
Clarify that a `[[registry.mirror]]` is associated only with the
previous `[[registry]]`.

Fixes: #1523
Signed-off-by: Valentin Rothberg <vrothberg@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
@mtrmac PTAL

**Comment by rhatdan** [comment]
LGTM
@TomSweeneyRedHat PTAL

File: docs/containers-registries.conf.5.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ as-is. But other settings like insecure/blocked/mirrors will be applied to match
 
 `mirror`
 : An array of TOML tables specifying (possibly-partial) mirrors for the
-`prefix`-rooted namespace.
+`prefix`-rooted namespace (i.e., the previous registry).
```
Comment (mtrmac) [inline docs/containers-registries.conf.5.md:102]: Is that the “current” registry, actually?

It’s not even a _“registry”_ at all, it’s that `prefix`-delineated space, which might be smaller than a registry.

How about “the current `[[registry]]` TOML table`?

File: docs/containers-registries.conf.5.md [hunk 2]
```diff
@@ -244,6 +250,8 @@ Given the above, a pull of `example.com/foo/image:latest` will try:
 
 in order, and use the first one that exists.
 
+Note that a mirror is associated only with the previous registry.  Pulling `registry.com/image:latest` will hence only reach out to `mirror.registry.com` and not use the one associated to the other registry.
```
Comment (mtrmac) [inline docs/containers-registries.conf.5.md:253]: Again, consider something like “`[[registry]]` TOML table” rather than just “registry” (in both sentences). Also `s/previous/current/` — `example.com/foo` is the “previous” one.

Or maybe say something to express the idea that “a [[registry.mirror]] TOML table is, per TOML semantics, logically nested inside the previous [[registry]] table, and only applies to that [[registry]] table” — to talk about the options and their nesting, instead of relative ordering of tables as if they were conceptually unrelated. I don’t have a good idea of which one is easier to understand.

Comment (mtrmac) [inline docs/containers-registries.conf.5.md:253]: Would it be clearer (or much less clear??) to explicitly list the [`"example-mirror-0.local"`, `"example-mirror-1.local"`] names instead of saying “the other registry”? Or maybe both, “and use neither `"example-mirror-0.local"` nor `"example-mirror-1.local"`, which only apply to the `example.com/foo` namespace”— OTOH that seems _way_ too long.

**Review by mtrmac — COMMENTED** [review]
Thanks!

Just some non-blocking suggestions to consider. I’ll fully defer to native speakers in all respects.

File: docs/containers-registries.conf.5.md [hunk 3]
```diff
@@ -236,6 +236,12 @@ location = "example-mirror-0.local/mirror-for-foo"
...
 insecure = true
+
+[[registry]]
+location = "registry.com"
+
+[[registry.mirror]]
+location = "mirror.registry.com"
 ```
 Given the above, a pull of `example.com/foo/image:latest` will try:
     1. `example-mirror-0.local/mirror-for-foo/image:latest`
```
Comment (TomSweeneyRedHat) [inline docs/containers-registries.conf.5.md:247]: Not part of this change, but it looks like there's an extra leading space on this line.

File: docs/containers-registries.conf.5.md (hunk 2 above)
Comment (vrothberg) [reply docs/containers-registries.conf.5.md:253]: Good idea!  We can keep it shorter by saying "the mirror of `example.com/foo`"

**Comment by vrothberg** [comment]
@mtrmac @TomSweeneyRedHat PTAL

**Review by mtrmac — APPROVED** [review]
LGTM. Thanks!

File: docs/containers-registries.conf.5.md (hunk 3 above)
Comment (mtrmac) [reply docs/containers-registries.conf.5.md:247]: Let’s fix that: https://github.com/containers/image/pull/1527 (_after_ this PR).

**Comment by rhatdan** [comment]
LGTM
